### PR TITLE
feat:  implement file view for publishing request wizard (Issue #331)

### PR DESCRIPTION
**Pull request by IlyaBondar**
- Issue #331 

### Based on
- PR #486

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

File: src/components/Chat/PublishModal.tsx [hunk 1]
```diff
@@ -153,136 +162,157 @@ export default function PublishModal({
...
-            </CollapsableSection>
-          ))}
-        </section>
-
-        <div className="flex justify-end gap-3">
+          {(type === SharingType.Conversation ||
+            type === SharingType.ConversationFolder) && (
+            <div className="flex w-full flex-col gap-3 p-4 md:max-w-[550px]">
+              <h2>
+                {t(`Files contained in the ${getPrefix(entity).toLowerCase()}`)}
```
Comment (denys-kolomiitsev) [inline src/components/Chat/PublishModal.tsx:326]: To use dynamic values in translations in the **i18next.t** we need to do it in this way

```suggestion
                {t(`Files contained in the {{prefix}}`, {prefix: getPrefix(entity).toLowerCase()})}
```
https://www.i18next.com/translation-function/interpolation

File: src/components/Chat/PublishModal.tsx [hunk 2]
```diff
@@ -153,136 +162,157 @@ export default function PublishModal({
...
+                  'Your conversation will be visible to organization only after verification by the administrator',
+                )}
+              </p>
+
+              <div>
+                <label
+                  className="mb-1 flex text-xs text-secondary"
+                  htmlFor="requestName"
+                >
+                  {t(`${getPrefix(entity)} name`)}
```
Comment (denys-kolomiitsev) [inline src/components/Chat/PublishModal.tsx:196]: To use dynamic values in translations in the **i18next.t** we need to do it in this way

       ```suggestion
                  {t('{{prefix}} name', {prefix: getPrefix(entity)})}
```

https://www.i18next.com/translation-function/interpolation

File: src/components/Chat/PublishModal.tsx (hunk 1 above)
Comment (IlyaBondar) [reply src/components/Chat/PublishModal.tsx:326]: @denys-kolomiitsev , it works only if we wan't interpolation in result string. Here I generate key and can use javascript interpolation

**Review by denys-kolomiitsev — DISMISSED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: failed

File: package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
     "eventsource-parser": "^0.1.0",
     "i18next": "^22.4.13",
     "isomorphic-dompurify": "^1.8.0",
+    "lodash-es": "^4.17.21",
```
Comment (mikitabut) [inline package.json]: Are you sure that we really need lodash?

File: src/components/Chat/PublishModal.tsx [hunk 4]
```diff
@@ -87,6 +103,33 @@ export default function PublishModal({
...
+      if (nameMap.has(oldPath)) {
+        const originalPath = nameMap.get(oldPath);
+        nameMap.delete(oldPath);
+        oldPath = originalPath;
+      }
+      newFileNames.current.set(newPath, oldPath);
+      file.name = name;
+      setRenamingFile(undefined);
+    },
+    [],
```
Comment (mikitabut) [inline src/components/Chat/PublishModal.tsx]: Warning here

File: src/types/share.ts [hunk 5]
```diff
@@ -21,4 +23,9 @@ export interface PublishRequest {
   name: string;
   path: string;
   version: string;
+  fileNameMapping: Map<string, string>;
+}
+
+export interface PublishAttachmentInfo extends Omit<DialFile, 'contentLength'> {
+  path: string;
```
Comment (mikitabut) [inline src/types/share.ts]: Why do you need one more path? Are relative path or absolute path not applicable for you?

File: src/components/Chat/PublishAttachment.tsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+            <Tooltip
+              tooltip={fullPath}
+              triggerClassName="block max-w-full truncate text-secondary"
+              hideTooltip={isContextMenu}
+            >
+              {fullPath}
+            </Tooltip>
+          </>
+        ) : (
+          <>
```
Comment (Alexander-Kezik) [inline src/components/Chat/PublishAttachment.tsx]: Do we need react fragment on 188 line?

File: src/utils/app/file.ts [hunk 7]
```diff
@@ -177,3 +184,61 @@ export const getExtensionsListForMimeTypes = (mimeTypes: string[]) => {
...
+      file !== renamingFile &&
+      file.path === renamingFile?.path,
+  );
+
+  if (fileWithSameName) {
+    return 'Not allowed to have folders with same names';
+  }
+
+  if (newName.match(notAllowedSymbolsRegex)) {
+    return `The symbols ${notAllowedSymbols} are not allowed in folder name`;
```
Comment (Alexander-Kezik) [inline src/utils/app/file.ts]: I think you meant a file, not a folder.

```suggestion
  if (fileWithSameName) {
    return 'Not allowed to have files with same names';
  }

  if (newName.match(notAllowedSymbolsRegex)) {
    return `The symbols ${notAllowedSymbols} are not allowed in file name`;
```

File: src/components/Chat/PublishModal.tsx [hunk 8]
```diff
@@ -153,136 +200,164 @@ export default function PublishModal({
   return (
     <Modal
       portalId="theme-main"
-      containerClassName="inline-block h-[747px] md:min-w-[550px] md:max-w-[1100px] p-6 group/modal"
+      containerClassName={classNames(
+        'group/modal inline-block h-[747px] max-h-full min-w-full max-w-[1100px] !bg-layer-2 md:min-w-[550px]',
```
Comment (Alexander-Kezik) [inline src/components/Chat/PublishModal.tsx]: `max-h-full` already in the Modal component

```suggestion
        'group/modal inline-block h-[747px] min-w-full max-w-[1100px] !bg-layer-2 md:min-w-[550px]',
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Review by mikitabut — DISMISSED** [review]

**Review by Alexander-Kezik — DISMISSED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**